pkg/probe/http/http.go

```diff
@@ -28,15 +29,18 @@ import (
 )
 
 func New() HTTPProber {
-	return HTTPProber{&http.Client{}}
+	client := &http.Client{}
+	return HTTPProber{client, func(timeout time.Duration) { client.Timeout = timeout }}
 }
 
 type HTTPProber struct {
-	client HTTPGetInterface
+	client     HTTPGetInterface
+	setTimeout func(timeout time.Duration)
 }
 
 // Probe returns a ProbeRunner capable of running an http check.
-func (pr *HTTPProber) Probe(host string, port int, path string) (probe.Status, error) {
+func (pr *HTTPProber) Probe(host string, port int, path string, timeout time.Duration) (probe.Status, error) {
+	pr.setTimeout(timeout)
```

Comment:
@commonlisp I looked over the net/http package last night in depth. It looks like http.Transport is the main reason http.Client is so expense (as it's caching tcp connections under the hood). If we had one http.Transport per probe runner, and created a client with `http.Client{Transport: runner.transport, Timeout: timeout}` we get the ease of use of the client. It also seems that if no transport is specified when a client is created, the http.DefaultTransport is used, so creating new http.Client{} isn't as expensive as I had previously thought. I think how it was before is reasonable, or if you want each HTTPProbeRunner to have it's own http.Transport. Sorry to make you jump through hoops. What do you think?